src/entity/object/FallingBlock.php

```diff
@@ -97,13 +104,27 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 		if($this->closed){
 			return false;
 		}
+		$this->fallTime++;
 
 		$hasUpdate = parent::entityBaseTick($tickDiff);
 
 		if(!$this->isFlaggedForDespawn()){
 			$world = $this->getWorld();
 			$pos = $this->location->add(-$this->size->getWidth() / 2, $this->size->getHeight(), -$this->size->getWidth() / 2)->floor();
 
+			if($this->getBlock()->getId() === VanillaBlocks::ANVIL()->getId()) {
```

Comment:
This would be better in `Anvil->tickFalling()`.

Reply:
On second look, this is in completely the wrong place. Anvil damage should only apply when the anvil entity actually turns back into a block.

Reply:
That is not the case for dripstone, it damages everything in it's path and i am pretty sure anvils also work this way.

Reply:
I will switch it but I feel like this is a case where the vanilla implementation is worse

Reply:
On a side note, standing on slabs does not prevent you from getting the damage